Index: Makefile.am
===================================================================
RCS file: /cvsroot/audacity/lib-src/libsndfile/Makefile.am,v
retrieving revision 1.7
diff -u -r1.7 Makefile.am
--- Makefile.am	23 May 2006 22:05:39 -0000	1.7
+++ Makefile.am	23 Jan 2009 22:27:02 -0000
@@ -1,5 +1,8 @@
 ## Process this file with automake to produce Makefile.in
 
+# tell aclocal that extra macros can be found in autotools/m4/
+ACLOCAL_AMFLAGS = -I M4
+
 SUBDIRS = man doc Win32 Octave src examples regtest tests
 DIST_SUBDIRS = $(SUBDIRS)
 EXTRA_DIST = reconfigure.mk acinclude.m4 libsndfile.spec.in \
Index: configure.ac
===================================================================
RCS file: /cvsroot/audacity/lib-src/libsndfile/configure.ac,v
retrieving revision 1.10
diff -u -r1.10 configure.ac
--- configure.ac	17 Jan 2009 21:50:38 -0000	1.10
+++ configure.ac	23 Jan 2009 22:27:20 -0000
@@ -1,18 +1,20 @@
 # Copyright (C) 1999-2006 Erik de Castro Lopo (erikd AT mega-nerd DOT com).
 
 dnl Require autoconf version
-AC_PREREQ(2.59)
+AC_PREREQ(2.63)
 
 AC_INIT([libsndfile],[1.0.17],[erikd@mega-nerd.com])
 AC_CONFIG_SRCDIR([src/sndfile.c])
 AC_CANONICAL_TARGET([])
 
-AM_INIT_AUTOMAKE($PACKAGE_NAME,$PACKAGE_VERSION)
+AM_INIT_AUTOMAKE([-Wall])
 dnl Audacity policy: don't enable automatic rebuild of configure et al if 
 dnl sources change
 AM_MAINTAINER_MODE([enabled])
 
 AC_CONFIG_HEADERS([src/config.h])
+dnl keep macros in M4/
+AC_CONFIG_MACRO_DIR([M4])
 
 AC_LANG([C])
 
@@ -35,6 +37,8 @@
 SHARED_VERSION_INFO="1:17:0"
 
 AC_PROG_CC
+AM_PROG_CC_C_O
+AC_PROG_CXX
 AM_PROG_LIBTOOL
 
 AC_CHECK_PROG(autogen, autogen, yes, no)
@@ -80,7 +84,7 @@
 # Finished checking, handle options.
 
 AC_ARG_ENABLE(experimental,
-	AC_HELP_STRING([--enable-experimental], [enable experimental code]))
+	AS_HELP_STRING([--enable-experimental],[enable experimental code]))
 
 EXPERIMENTAL_CODE=0
 if test x$enable_experimental = xyes ; then
@@ -89,28 +93,28 @@
 AC_DEFINE_UNQUOTED([ENABLE_EXPERIMENTAL_CODE],${EXPERIMENTAL_CODE}, [Set to 1 to enable experimental code.])
 
 AC_ARG_ENABLE(gcc-werror,
-	AC_HELP_STRING([--enable-gcc-werror], [enable -Werror in all Makefiles]))
+	AS_HELP_STRING([--enable-gcc-werror],[enable -Werror in all Makefiles]))
 
 AC_ARG_ENABLE(gcc-pipe,
-	AC_HELP_STRING([--disable-gcc-pipe], [disable gcc -pipe option]))
+	AS_HELP_STRING([--disable-gcc-pipe],[disable gcc -pipe option]))
 
 AC_ARG_ENABLE(gcc-opt,
-	AC_HELP_STRING([--disable-gcc-opt], [disable gcc optimisations]))
+	AS_HELP_STRING([--disable-gcc-opt],[disable gcc optimisations]))
 
 AC_ARG_ENABLE(cpu-clip,
-	AC_HELP_STRING([--disable-cpu-clip], [disable tricky cpu specific clipper]))
+	AS_HELP_STRING([--disable-cpu-clip],[disable tricky cpu specific clipper]))
 
 AC_ARG_ENABLE(bow-docs,
-	AC_HELP_STRING([--enable-bow-docs], [enable black-on-white html docs]))
+	AS_HELP_STRING([--enable-bow-docs],[enable black-on-white html docs]))
 
 AC_ARG_ENABLE(sqlite,
-	AC_HELP_STRING([--disable-sqlite], [disable use of sqlite]))
+	AS_HELP_STRING([--disable-sqlite],[disable use of sqlite]))
 
 AC_ARG_ENABLE(flac,
-	AC_HELP_STRING([--disable-flac], [disable use of FLAC]))
+	AS_HELP_STRING([--disable-flac],[disable use of FLAC]))
 
 AC_ARG_ENABLE(alsa,
-	AC_HELP_STRING([--disable-alsa], [disable use of ALSA]))
+	AS_HELP_STRING([--disable-alsa],[disable use of ALSA]))
 
 #====================================================================================
 # Check types and their sizes.
Index: examples/Makefile.am
===================================================================
RCS file: /cvsroot/audacity/lib-src/libsndfile/examples/Makefile.am,v
retrieving revision 1.6
diff -u -r1.6 Makefile.am
--- examples/Makefile.am	23 May 2006 22:06:40 -0000	1.6
+++ examples/Makefile.am	23 Jan 2009 22:27:28 -0000
@@ -12,7 +12,7 @@
 OS_SPECIFIC_LINKS = @OS_SPECIFIC_LINKS@
 
 SNDFILEDIR =../src
-INCLUDES = -I$(srcdir)/$(SNDFILEDIR) $(OS_SPECIFIC_CFLAGS)
+AM_CPPFLAGS = -I$(srcdir)/$(SNDFILEDIR) $(OS_SPECIFIC_CFLAGS)
 
 sndfile_info_SOURCES = sndfile-info.c 
 sndfile_info_LDADD = $(SNDFILEDIR)/libsndfile.la
Index: regtest/Makefile.am
===================================================================
RCS file: /cvsroot/audacity/lib-src/libsndfile/regtest/Makefile.am,v
retrieving revision 1.4
diff -u -r1.4 Makefile.am
--- regtest/Makefile.am	2 Feb 2008 16:39:05 -0000	1.4
+++ regtest/Makefile.am	23 Jan 2009 22:27:29 -0000
@@ -5,7 +5,7 @@
 noinst_HEADERS = regtest.h
 
 SNDFILEDIR =../src
-INCLUDES = -I$(srcdir)/$(SNDFILEDIR) $(OS_SPECIFIC_CFLAGS)
+AM_CPPFLAGS = -I$(srcdir)/$(SNDFILEDIR) $(OS_SPECIFIC_CFLAGS)
 
 sndfile_regtest_SOURCES = sndfile-regtest.c database.c checksum.c
 sndfile_regtest_LDADD = $(SNDFILEDIR)/libsndfile.la $(SQLITE3_LIBS)
Index: tests/Makefile.am
===================================================================
RCS file: /cvsroot/audacity/lib-src/libsndfile/tests/Makefile.am,v
retrieving revision 1.10
diff -u -r1.10 Makefile.am
--- tests/Makefile.am	2 Feb 2008 16:39:07 -0000	1.10
+++ tests/Makefile.am	23 Jan 2009 22:27:31 -0000
@@ -9,7 +9,7 @@
 	locale_test largefile_test win32_ordinal_test cpp_test
 
 SNDFILEDIR = ../src
-INCLUDES = -I$(srcdir)/$(SNDFILEDIR)
+AM_CPPFLAGS = -I$(srcdir)/$(SNDFILEDIR)
 noinst_HEADERS = dft_cmp.h utils.h
 
 autogen_sources = write_read_test.tpl write_read_test.def	\
