--- ../../../portaudio/configure.in	2007-12-02 17:25:17.000000000 +0000
+++ portaudio-v19/configure.in	2007-12-09 20:15:25.000000000 +0000
@@ -42,6 +42,10 @@
             [  --with-macapi ((asio/core/sm) default=core)],
             with_macapi=$withval, with_macapi="core")
 
+AC_ARG_WITH(universal_binary,
+            [  --with-universal-binary (Both Mac PPC and Intel, default=no)],
+            with_universal_binary=$withval, with_universal_binary="no")
+
 AC_ARG_WITH(asiodir,
             [  --with-asiodir (default=/usr/local/asiosdk2)],
             with_asiodir=$withval, with_asiodir="/usr/local/asiosdk2")
@@ -134,17 +138,48 @@
    CFLAGS="$CFLAGS -DPA_LITTLE_ENDIAN"
 fi
 
+MAC_BEFORE_TIGER="no"
+case "${host_os}" in 
+  darwin6* )
+	echo "We're running on Mac OS X 10.2"
+	MAC_BEFORE_TIGER="yes"
+	;;
+  darwin7* )
+	echo "We're running on Mac OS X 10.3"
+	MAC_BEFORE_TIGER="yes"
+	;;
+esac
+
+case "${NEXT_ROOT}" in
+  /Developer/SDKs/MacOSX10.2* )
+	echo "We're compiling for Mac OS X 10.2"
+	MAC_BEFORE_TIGER="yes"
+	;;
+  /Developer/SDKs/MacOSX10.3* )
+	echo "We're compiling for Mac OS X 10.3"
+	MAC_BEFORE_TIGER="yes"
+	;;
+esac
+
 case "${host_os}" in
   darwin* )
 	dnl Mac OS X configuration
 
 	AC_DEFINE(PA_USE_COREAUDIO)
+	dnl the SHARED_FLAGS needed regardless of 10.5 or not, and regardless of
+	dnl Universal Binary or not.
+	SHARED_FLAGS="-framework CoreAudio -framework AudioToolbox -framework AudioUnit -framework Carbon -dynamiclib -Werror"
 	if [[ -d /Developer/SDKs/MacOSX10.5.sdk ]] ; then
-		SHARED_FLAGS="-Werror -framework CoreAudio -framework AudioToolbox -framework AudioUnit -framework Carbon -dynamiclib -arch x86_64 -arch ppc64 -arch i386 -arch ppc -isysroot /Developer/SDKs/MacOSX10.5.sdk -mmacosx-version-min=10.3";
-		CFLAGS="-Werror $CFLAGS -arch x86_64 -arch ppc64 -arch i386 -arch ppc -isysroot /Developer/SDKs/MacOSX10.5.sdk -mmacosx-version-min=10.3";
+		dnl this is stuff that is 10.5-specific
+		dnl extra SHARED_FLAGS needed for 10.5
+		UNIV_SHARED_FLAGS="-arch x86_64 -arch ppc64 -isysroot /Developer/SDKs/MacOSX10.5.sdk";
+		dnl these extra CFLAGs are only needed for Universal Binary support
+		dnl when building on 10.5 and above
+		UNIV_CFLAGS="-arch x86_64 -arch ppc64";
 	else
-		SHARED_FLAGS="-Werror -framework CoreAudio -framework AudioToolbox -framework AudioUnit -framework Carbon -dynamiclib  -arch i386 -arch ppc -isysroot /Developer/SDKs/MacOSX10.4u.sdk -mmacosx-version-min=10.3";
-		CFLAGS="-Werror $CFLAGS -arch i386 -arch ppc -isysroot /Developer/SDKs/MacOSX10.4u.sdk -mmacosx-version-min=10.3";
+		dnl anything except 10.5 goes here
+		UNIV_SHARED_FLAGS="-isysroot /Developer/SDKs/MacOSX10.4u.sdk";
+		UNIV_CFLAGS="";
 	fi
 	OTHER_OBJS="src/os/mac_osx/pa_mac_hostapis.o src/os/unix/pa_unix_util.o src/hostapi/coreaudio/pa_mac_core.o src/hostapi/coreaudio/pa_mac_core_utilities.o src/hostapi/coreaudio/pa_mac_core_blocking.o src/common/pa_ringbuffer.o";
 	LIBS="-framework CoreAudio -framework AudioToolbox -framework AudioUnit -framework Carbon";
@@ -161,6 +196,23 @@
             CFLAGS="$CFLAGS -I\$(top_srcdir)/pa_asio -I$ASIDIR/host/mac -I$ASIODIR/common";
             CXXFLAGS="$CFLAGS";
         fi
+
+	if [[ $with_universal_binary = "yes" ]] ; then
+	   dnl these are the CFLAGs needed for Universal Binary builds on
+	   dnl all OS X targets. $UNIV_CFLAGS contains extra CFLAGS needed for
+	   dnl Universal. Content depends on if 10.5 SDK is installed. Flags here
+	   dnl are needed for universal regardless of 10.5 or not
+	   CFLAGS="${CFLAGS} ${UNIV_CFLAGS} -arch i386 -arch ppc -isysroot /Developer/SDKs/MacOSX10.4u.sdk -mmacosx-version-min=10.3 -Werror"
+	   dnl ${UNIV_SHARED_FLAGS} is the extra flags for universal binary builds
+	   dnl content depends on the OS X version in use. Flags listed here are
+	   dnl needed for Universal Binary regardless of the OS X version
+	   SHARED_FLAGS="${SHARED_FLAGS} ${UNIV_SHARED_FLAGS} -arch i386 -arch ppc -mmacosx-version-min=10.3";
+	fi
+
+	if [[ $MAC_BEFORE_TIGER = "yes" ]] ; then
+	    echo "Disabling SMP memory protection because we're on Mac OS X prior to 10.4 (Tiger)."
+	    CFLAGS="$CFLAGS -DALLOW_SMP_DANGERS=1"
+	fi
 	;;
 
   mingw* )
