--- a/lib-src/libvamp/vamp-sdk/PluginAdapter.cpp
+++ b/lib-src/libvamp/vamp-sdk/PluginAdapter.cpp
@@ -36,6 +36,9 @@
 
 #include "PluginAdapter.h"
 
+#include <cstdlib>
+#include <cstring>
+
 //#define DEBUG_PLUGIN_ADAPTER 1
 
 
--- a/lib-src/libvamp/vamp-sdk/PluginHostAdapter.cpp
+++ b/lib-src/libvamp/vamp-sdk/PluginHostAdapter.cpp
@@ -36,6 +36,8 @@
 
 #include "PluginHostAdapter.h"
 
+#include <cstdlib>
+
 namespace Vamp
 {
 
--- a/lib-src/libvamp/vamp-sdk/hostext/PluginLoader.cpp
+++ b/lib-src/libvamp/vamp-sdk/hostext/PluginLoader.cpp
@@ -61,6 +61,8 @@
 
 #endif /* ! _WIN32 */
 
+#include <cstring>
+
 using namespace std;
 
 namespace Vamp {
